src/Aspire.Hosting.Azure/Provisioning/Internal/DefaultProvisioningContextProvider.cs

```diff
@@ -97,54 +90,160 @@ private async Task RetrieveAzureProvisioningOptions(JsonObject userSecrets, Canc
+
+                var azureSection = userSecrets.Prop("Azure");
+
+                // Persist the parameter value to user secrets so they can be reused in the future
+                azureSection["SubscriptionId"] = _options.SubscriptionId;
+                azureSection["Location"] = _options.Location;
+                azureSection["ResourceGroup"] = _options.ResourceGroup;
+
+                _provisioningOptionsAvailable.SetResult();
+            }
+        }
+    }
+
+    private async Task<string?> PromptForSubscriptionAsync(CancellationToken cancellationToken = default)
+    {
+        // Get the available subscriptions based on the current credential or fallback to text input
+        try
+        {
+            var credential = tokenCredentialProvider.TokenCredential;
+            var armClient = armClientProvider.GetArmClient(credential);
+
+            logger.LogInformation("Fetching available Azure subscriptions...");
+            var availableSubscriptions = await armClient.GetAvailableSubscriptionsAsync(cancellationToken).ConfigureAwait(false);
+
+            if (availableSubscriptions.Count == 0)
+            {
+                throw new InvalidOperationException("No Azure subscriptions are available for the authenticated user.");
+            }
+
+            var subscriptionOptions = availableSubscriptions
+                .Select(s => KeyValuePair.Create(s.SubscriptionId, $"{s.DisplayName} ({s.SubscriptionId})"))
+                .ToList();
+
+            var result = await interactionService.PromptInputsAsync(
+                AzureProvisioningStrings.SubscriptionInputsTitle,
+                AzureProvisioningStrings.SubscriptionInputsMessage,
+                [
+                    new InteractionInput
+                    {
+                        InputType = InputType.Choice,
+                        Label = AzureProvisioningStrings.SubscriptionLabel,
+                        Placeholder = AzureProvisioningStrings.SubscriptionPlaceholder,
+                        Required = true,
+                        Options = [..subscriptionOptions]
+                    }
+                ],
+                new InputsDialogInteractionOptions
+                {
+                    EnableMessageMarkdown = true
+                },
+                cancellationToken).ConfigureAwait(false);
+
+            return result.Canceled ? null : result.Data?[0].Value;
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Failed to fetch Azure subscriptions. Falling back to manual subscription ID entry.");
+
+            var result = await interactionService.PromptInputsAsync(
+                AzureProvisioningStrings.SubscriptionInputsTitle,
+                "Unable to automatically fetch subscriptions. Please enter your subscription ID manually.",
+                [
+                    new InteractionInput
+                    {
+                        InputType = InputType.SecretText,
```

Comment:
Is SubscriptionId really a secret? If so, we shouldn't be displaying them in the choice list above.

IMO I think we can make this a regular text box.